src/test/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/model/OffenceSexualOrViolentTest.kt

```diff
@@ -1,64 +1,48 @@
 package uk.gov.justice.digital.hmpps.manageoffencesapi.model
 
-import org.junit.jupiter.api.Assertions.assertEquals
-import org.junit.jupiter.api.Test
-import uk.gov.justice.digital.hmpps.manageoffencesapi.model.OffenceSexualOrViolentIndicator.NONE
-import uk.gov.justice.digital.hmpps.manageoffencesapi.model.OffenceSexualOrViolentIndicator.SEXUAL
-import uk.gov.justice.digital.hmpps.manageoffencesapi.model.OffenceSexualOrViolentIndicator.VIOLENT
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
 
 class OffenceSexualOrViolentTest {
 
-  @Test
-  fun `Test all combinations of schedules return the correct result`() {
-    assertEquals(
-      SEXUAL,
+  @ParameterizedTest
+  @CsvSource(
+//   S3     S15P1  S15P2  toggle  code     result
+    "true,  false, false, false,  AB1234,  SEXUAL",
+    "false, false, true,  false,  AB1234,  SEXUAL",
+    "false, true,  false, false,  AB1234,  VIOLENT",
+    "false, false, false, false,  AB1234,  NONE",
+//   Sexual takes priority over violent
+    "true , true,  false, false,  AB1234,  SEXUAL",
+    "false, true,  true,  false,  AB1234,  SEXUAL",
+//   Tests for the toggle
+    "false, false, false, true,   SX0334,  SEXUAL",
+    "false, false, false, true,   SX5634B, SEXUAL",
+    "false, false, false, true,   AB5634B, NONE",
+    "false, false, true,  true,   AB5634B, SEXUAL",
+    "false, true,  false, true,   AB5634B, VIOLENT",
+//   Sexual takes priority over violent (with codes)
+    "false, true,  false, true,   SX0345,  SEXUAL",
+    "false, true,  false, true,   SX5678,  SEXUAL",
+    "false, true,  true,  true,   CF2478,  SEXUAL",
+  )
```

Comment:
nice

Reply:
I picked it up from you, managed to fit 4 x tests in half the number of lines